framework/elements/tree/function_to_tree.py

```diff
@@ -0,0 +1,195 @@
+
+
+
+class StringToTree:
+
+    parenMarker = 0  # Helps indicates if a substring is surrounded by open and closed parentheses
+
+    parenIndex = -1  # Marks positions of parenthesis in the function string
+
+    cropIndex = [0, 0]  # Indicates the positions of two corresponding open and closed parentheses
+
+    firstTreeList = []  # Inital representation of function tree
+
+    subLists = []  # Stores sublist for recursion
+
+    operands = {"prod": "*", "sum": "+", "diff": "-", "quot": "/", "mod": "%"}  # Acts as a operator bank 
+
+    finalTreeList = []  # Final (Level Traversed) representation of function tree 
+
+    kidslist = []  # Stores kids of each level traversed node
+
+    '''String to tree:
+        This class uses three functions to convert from a function to a usable
+        tree for evolf based on the function paramenters.
+
+        StringToTree1: This turns a function to a list that contains nodes 
+        represented as "[node function, level, children]" where the level 
+        is a marker of how deep a node is in the function
+
+        StringToTree2: This turns the output from the above function into a 
+        similar function with null nodes included
+
+        StringToTree3: This turns the output from the above function above 
+        into a level trvaversed list of function names.
+
+        To make use of the class use the function StringToTree which combines
+        all three of the functions above.
+              '''
+    @classmethod
+    def stringToTree1(cls, string, level=0):
+
```

Comment:
For function descriptions add the comments here explaining what it does rather than up top. The comment at the top just describes the purpose of the class and maybe some usage and edge cases